feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/overview/webViewController/WebViewCardCreationInterceptor.kt

```diff
@@ -0,0 +1,135 @@
+) {
+
+    private var interceptedCardRequest = MutableStateFlow<Request.Builder?>(null)
+
+    fun runPolling(coroutineScope: CoroutineScope) {
+        interceptedCardRequest.onEachLatest { request ->
+            if (request == null) return@onEachLatest
+
+            for (i in 0..100) {
+                val cardWasSuccessfullyCreated = makeRequest(request)
+
+                if (cardWasSuccessfullyCreated) {
+                    break
+                }
+
+                delay(1000.milliseconds)
+            }
+        }.inBackground()
+            .launchIn(coroutineScope)
+    }
+
+    /**
+     * @return true if the request was intercepted otherwise false
+     */
+    fun intercept(request: WebResourceRequest): Boolean {
+        val url = request.url.toString()
+
+        // Intercept requests that might return a JSON response
+        if (url.contains("https://api.mercuryo.io/v1.6/cards")) { // Specify your condition here
+            performOkHttpRequest(request)
+            return true
+        }
+
+        return false
+    }
+
+    private fun performOkHttpRequest(request: WebResourceRequest) {
+        try {
+            // Create OkHttp Request based on WebResourceRequest
+            val okHttpRequestBuilder = Request.Builder().url(request.url.toString())
+
+            // Set method (GET, POST, etc.) and request body if needed
+            when (request.method) {
+                "GET" -> okHttpRequestBuilder.get()
+                else -> okHttpRequestBuilder.get()
+            }
+
+            // Add headers from WebResourceRequest
+            for ((key, value) in request.requestHeaders) {
+                okHttpRequestBuilder.addHeader(key, value)
+            }
+
+            val cookieManager = CookieManager.getInstance()
+            val cookies = cookieManager.getCookie(request.url.toString())
+            if (cookies != null) {
+                okHttpRequestBuilder.addHeader("Cookie", cookies)
+            }
+
+            interceptedCardRequest.value = okHttpRequestBuilder
+
+            makeRequest(okHttpRequestBuilder)
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+    }
+
+    /**
+     * Notify onCardCreatedListener if data value contains mercurio cards
+     */
+    private fun makeRequest(requestBuilder: Request.Builder): Boolean {
+        // Execute the OkHttp request
+        val okHttpResponse = okHttpClient.newCall(requestBuilder.build()).execute()
+
+        // Check if the response is successful
+        if (okHttpResponse.isSuccessful) {
+            val responseBody = okHttpResponse.body
+
+            val data = responseBody!!.byteStream().readText()
+            val cardsResponse = gson.fromJson<CardsResponse>(data)
+            val cards = cardsResponse.data
+            val containsMercurioCard = cards.any { it.issuedByMercurio }
+
+            if (containsMercurioCard) {
+                onCardCreatedListener.onCardCreated()
```

Comment:
This is an unexpected side effect, you wont expect `makeRequest` to notify the listener
Maybe we should separate `makeRequest` and listener notification? E.g. makeRequest already returns true, upper-level logic can use that to determine whether to invoke the listener or not, this method can be called something like `checkForCardCreation`